File: app/pages/explore.py
from flask import render_template, request, redirect
from app import app
from app.database import *

import json


@app.route('/explore', methods=['GET', 'POST'])
def hello_explore():
    if request.method == 'GET':
        username = request.args.get('username')
        if username is None:
            # return redirect('/')
            pass
        return render_template("explore.html", username="123")
    else:
        data = json.loads(request.get_data())
        print(data)
        dtype = data['type']
        if dtype == "recommend":
            # 推荐用户
            uid = data['username']
            user = User.get(uid)
            ans = user.recommend(3)
            result = []
            for rec in ans:
                assert isinstance(rec, User)
                result.append({
                    "username": rec.uid,
                    "nickname": rec.nickname,
                    "portrait": rec.portrait
                })
            # print(result)
            return json.dumps(result)

        elif dtype == "explore":
            uid = data['username']
            ans = Album.explore(uid, 9)

            result = []
            for al in ans:
                assert isinstance(al, Album)
                result.append({
                    "nickname": User.get(al.uid).nickname,
                    "img": al.photo
                })
            # print(result)
            return json.dumps(result)

        elif dtype == "view":
            un = data['username']
            print(un)
            return "123"

        else:
            return None

File: app/static/js/explore.js
var user_update = function() {
    var data = JSON.stringify({
        type: "recommend",
        username: username,
    })
    $.ajax({
        data: data,
        dataType: "json",
        type: "POST",
        success: function(result) {
            var users = eval(result);
            var i = 0;
            $(".user_detail").each(function(){
                if(users[i] == undefined) return false;
                $(this).find('img').attr('src', users[i].portrait);
                $(this).find('img').attr('href', "/explore?username=" + users[i].username)
                $(this).find('.nickname').html(users[i].nickname)
                $(this).find('.uid').html(users[i].username)
                i++;
            });
        }
    })
}

var thumb_update = function() {
    var data = JSON.stringify({
        type: "explore",
        username: username
    })
    $.ajax({
        data: data,
        dataType: "json",
        type: "POST",
        success: function(result) {
            var images = eval(result)//返回的随机图片组
            var i = 0;
            $(".thumb").each(function(){
                if(images[i] == undefined) return false;
                $(this).find('a').attr('href', images[i].img)
                $(this).find('h1').html(images[i].nickname+"<em>’s photo</em>")
                $(this).find('img').attr('src', images[i].img)
                i++;
            });
        }
    })
}

user_update();
thumb_update();

$(function(){

    $("#change-button").click(function(){
        user_update();
        thumb_update();
    })

    $("#search-form").submit(function(){
        var e = $("#explorer").val();
        if(e == "") return false
        window.location.href="homepage?"+"username="+encodeURI(e)+"&explorer="+encodeURI(username);//change after merge
        return false
    })

    $("#homepage").click(function(){
        window.location.href="homepage?"+"username="+encodeURI(username)+"&explorer="+encodeURI(username);//change after merge

    })

    $("#explore").click(function(){
        window.location.href="explore?"+"username="+encodeURI(username);//change after merge

    })

    $(".view-button").click(function(){
        var target = $(this).parents().find(".uid").html()

        window.location.href="homepage?username="+encodeURI(target)+"&explorer="+encodeURI(username);//change after merge
    })

})